src/jit/lclvars.cpp

```diff
@@ -488,6 +488,26 @@ void Compiler::lvaInitRetBuffArg(InitVarDscInfo* varDscInfo)
                 varDsc->lvType = TYP_I_IMPL;
             }
         }
+#ifdef FEATURE_SIMD
+        else if (featureSIMD && varTypeIsSIMD(info.compRetType))
+        {
+            CORINFO_SIG_INFO sigInfo;
+            info.compCompHnd->getMethodSig(info.compMethodHnd, &sigInfo);
+
+            // If the return buffer is a SIMD type, we need to note that by setting lvSIMDType.
+            var_types simdBaseType = TYP_UNKNOWN;
+            var_types type         = impNormStructType(sigInfo.retTypeClass, nullptr, nullptr, &simdBaseType);
```

Comment:
We have info.compMethodInfo->args.retTypeClass gives the class handle of the return type, there is no need to call getMethodSig() VM call?

I see impNormStructType() call for SIMD type struct return in lvaInitTypeRef() and it would be redundant here.  I think here we can call

varDsc->lvSIMDType = true
varDsc->lvBaseType = getBaseTypeAndSizeOfSIMDType(info.compMethodInfo->args.retTypeClass, &varDsc->lvExactSize);


Given info.compRetType is a SIMD type, its base type cannot be TYP_UNKNOWN.  In fact we should assert it here rather than checking.